js/packages/web/src/components/AuctionCard/index.tsx

```diff
@@ -516,74 +520,132 @@ export const AuctionCard = ({
 
               const instantSale = async () => {
                 setLoading(true);
-                const instantSalePrice =
-                  auctionView.auctionDataExtended?.info.instantSalePrice;
-                const winningConfigType =
-                  auctionView.items[0][0].winningConfigType;
-                const isAuctionItemMaster =
-                  winningConfigType === WinningConfigType.FullRightsTransfer ||
-                  winningConfigType === WinningConfigType.TokenOnlyTransfer;
-                const allowBidToPublic =
-                  myPayingAccount &&
-                  !auctionView.myBidderPot &&
-                  isAuctionManagerAuthorityNotWalletOwner;
-                const allowBidToAuctionOwner =
-                  myPayingAccount &&
-                  !isAuctionManagerAuthorityNotWalletOwner &&
-                  isAuctionItemMaster;
-
-                // Placing a "bid" of the full amount results in a purchase to redeem.
-                if (
-                  instantSalePrice &&
-                  (allowBidToPublic || allowBidToAuctionOwner)
-                ) {
+
+                try {
+                  const instantSalePrice =
+                    auctionView.auctionDataExtended?.info.instantSalePrice;
+                  const winningConfigType =
+                    auctionView.items[0][0].winningConfigType;
+                  const isAuctionItemMaster =
+                    winningConfigType ===
+                      WinningConfigType.FullRightsTransfer ||
+                    winningConfigType === WinningConfigType.TokenOnlyTransfer;
+                  const allowBidToPublic =
+                    myPayingAccount &&
+                    !auctionView.myBidderPot &&
+                    isAuctionManagerAuthorityNotWalletOwner;
+                  const allowBidToAuctionOwner =
+                    myPayingAccount &&
+                    !isAuctionManagerAuthorityNotWalletOwner &&
+                    isAuctionItemMaster;
+
+                  // Placing a "bid" of the full amount results in a purchase to redeem.
+                  if (
+                    instantSalePrice &&
+                    (allowBidToPublic || allowBidToAuctionOwner)
+                  ) {
+                    let bidTxid: string | undefined;
+
+                    try {
+                      const { amount, txid } = await sendPlaceBid(
+                        connection,
+                        wallet,
+                        myPayingAccount.pubkey,
+                        auctionView,
+                        accountByMint,
+                        instantSalePrice,
+                        'confirmed',
+                      );
+                      setLastBid({ amount });
+                      bidTxid = txid;
+                    } catch (e) {
+                      console.error('sendPlaceBid', e);
+                      return;
+                    }
+
+                    try {
```

Comment:
Note for use to ship this as a helper function in in the connection context:

https://github.com/metaplex-foundation/metaplex/blob/c92da34bd09dd56b4a622ea4f6a7867555425e95/js/packages/common/src/contexts/connection.tsx